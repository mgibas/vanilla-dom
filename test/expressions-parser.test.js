const parser = require('../src/expressions-parser')

describe('expressions-parses', () => {
  describe('on parse', () => {
    it('returns undefined when no expressions', () => {
      expect(parser.parse('')).not.toBeDefined();
    })

    it.each([
      ['{state.fullname}'], 
      ['{state.firstname + state.lastname}'], 
      ['{state.fullname.toString()}'],
      ['{new Date(state.year, state.month, state.day).toString()}'],
      ['{state.firstname} and {state.lastname}']
    ])('return propper expression descriptor for %s',
      (expression) => {
        expect(parser.parse(expression)).toMatchSnapshot();
      }
    );
  })
})
