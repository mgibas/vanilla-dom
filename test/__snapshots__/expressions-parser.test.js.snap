// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expressions-parses on parse return propper expression descriptor for {new Date(state.year, state.month, state.day).toString()} 1`] = `
Object {
  "paths": Array [
    "year",
    "month",
    "day",
  ],
  "template": "\`\${(() => {
        try {
          return new Date(state.year, state.month, state.day).toString()
        } catch (err) {
          return ''
        }
      })() || ''}\`",
}
`;

exports[`expressions-parses on parse return propper expression descriptor for {state.firstname + state.lastname} 1`] = `
Object {
  "paths": Array [
    "firstname",
    "lastname",
  ],
  "template": "\`\${(() => {
        try {
          return state.firstname + state.lastname
        } catch (err) {
          return ''
        }
      })() || ''}\`",
}
`;

exports[`expressions-parses on parse return propper expression descriptor for {state.firstname} and {state.lastname} 1`] = `
Object {
  "paths": Array [
    "firstname",
    "lastname",
  ],
  "template": "\`\${(() => {
        try {
          return state.firstname
        } catch (err) {
          return ''
        }
      })() || ''} and \${(() => {
        try {
          return state.lastname
        } catch (err) {
          return ''
        }
      })() || ''}\`",
}
`;

exports[`expressions-parses on parse return propper expression descriptor for {state.fullname.toString()} 1`] = `
Object {
  "paths": Array [
    "fullname",
  ],
  "template": "\`\${(() => {
        try {
          return state.fullname.toString()
        } catch (err) {
          return ''
        }
      })() || ''}\`",
}
`;

exports[`expressions-parses on parse return propper expression descriptor for {state.fullname} 1`] = `
Object {
  "paths": Array [
    "fullname",
  ],
  "template": "\`\${(() => {
        try {
          return state.fullname
        } catch (err) {
          return ''
        }
      })() || ''}\`",
}
`;
